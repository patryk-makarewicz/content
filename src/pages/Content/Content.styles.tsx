import styled from 'styled-components';
import { colors } from 'styles';

import { fontWeight, padding } from 'styles/variables';

export const Container = styled.div`
  width: 100%;
  min-height: calc(100vh - 155px);
`;

export const Paragraph = styled.p`
  text-align: center;
`;

export const ErrorMessage = styled(Paragraph)`
  font-weight: ${fontWeight.semiBold};
  color: ${colors.warning};
`;

export const Box = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100%;
  padding: ${padding.xl} 0;
`;
