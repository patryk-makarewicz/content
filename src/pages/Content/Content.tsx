import { useTranslation } from 'react-i18next';

import { useGetContent } from 'hooks/useGetContent';

import { Article } from 'components/Article';
import { Spinner } from 'components/Spinner';
import { SectionTitle } from 'components/Typography/SectionTitle';

import { PageLayout } from 'layout/PageLayout';

import * as Styled from './Content.styles';

export const Content = () => {
  const { t } = useTranslation();

  const { article, isLoading, isError } = useGetContent();

  return (
    <Styled.Container>
      <PageLayout>
        <SectionTitle title={t('pages.content.title')} />
        <Styled.Paragraph>{t('pages.content.description')}</Styled.Paragraph>
        <Styled.Box>
          {isLoading && <Spinner />}
          {isError && <Styled.ErrorMessage>{t('messages.fail.generic')}</Styled.ErrorMessage>}
          {!isLoading && !isError && <Article article={article} />}
        </Styled.Box>
      </PageLayout>
    </Styled.Container>
  );
};
