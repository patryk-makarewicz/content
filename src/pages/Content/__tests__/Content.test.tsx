import { render } from '@testing-library/react';
import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { RecoilRoot } from 'recoil';

import { Content } from '../Content';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const renderContent = () => {
  const { asFragment, getByTestId } = render(
    <RecoilRoot>
      <QueryClientProvider client={queryClient}>
        <Content />
      </QueryClientProvider>
    </RecoilRoot>
  );

  return {
    asFragment,
    getByTestId,
  };
};

describe('test Content', () => {
  it('should take and matches the snapshot', () => {
    const { asFragment } = renderContent();
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });

  it('should render title and spinner onLoad data', () => {
    const { getByTestId } = renderContent();

    const title = getByTestId('section-title');
    expect(title).toBeInTheDocument();

    const spinner = getByTestId('spinner');
    expect(spinner).toBeInTheDocument();
  });
});
