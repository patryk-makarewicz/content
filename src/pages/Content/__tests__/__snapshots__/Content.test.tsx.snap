// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Content should take and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jlZhew dxCNyp"
  >
    <section
      class="sc-dAlyuH bZWaEy"
    >
      <h2
        class="sc-kpDqfm gWZOTg"
        data-testid="section-title"
      >
        pages.content.title
      </h2>
      <p
        class="sc-cwHptR izyYR"
      >
        pages.content.description
      </p>
      <div
        class="sc-cPiKLX kkZFtw"
      >
        <div
          class="sc-jXbUNg hQcIHN"
        >
          <svg
            class="sc-dhKdcB dtPfqK spinner"
            data-testid="spinner"
            height="50px"
            version="1.1"
            viewBox="0 0 80 80"
            width="50px"
            x="0px"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
            y="0px"
          >
            <path
              d="M40,72C22.4,72,8,57.6,8,40C8,22.4, 22.4,8,40,8c17.6,0,32,14.4,32,32c0,1.1-0.9,2-2,2 s-2-0.9-2-2c0-15.4-12.6-28-28-28S12,24.6,12,40s12.6, 28,28,28c1.1,0,2,0.9,2,2S41.1,72,40,72z"
              fill="#1677ff"
              id="spinner"
            />
          </svg>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
